exp/controllers/azuremanagedcontrolplane_reconciler.go

```diff
@@ -0,0 +1,128 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+	"sigs.k8s.io/cluster-api/util/secret"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+
+	"sigs.k8s.io/cluster-api-provider-azure/azure"
+	"sigs.k8s.io/cluster-api-provider-azure/azure/scope"
+	"sigs.k8s.io/cluster-api-provider-azure/azure/services/groups"
+	"sigs.k8s.io/cluster-api-provider-azure/azure/services/managedclusters"
+	"sigs.k8s.io/cluster-api-provider-azure/azure/services/subnets"
+	"sigs.k8s.io/cluster-api-provider-azure/azure/services/virtualnetworks"
+	"sigs.k8s.io/cluster-api-provider-azure/util/tele"
+)
+
+// azureManagedControlPlaneService contains the services required by the cluster controller.
+type azureManagedControlPlaneService struct {
+	kubeclient         client.Client
+	scope              managedclusters.ManagedClusterScope
+	managedClustersSvc *managedclusters.Service
```

Comment:
nit: I guess making managedClusterSvc of type azure.Reconciler would be better, it would align it with other services? 

What do you think ? Any specific reason not using azure.Reconciler ?

Reply:
I agree, fixed it to use `azure.Reconciler`.